scripts/firmware.py

```diff
@@ -163,14 +203,22 @@ def colored_text(text: str, color: Color) -> str:
 
     @staticmethod
     def error_msg(message: str) -> None:
-        print(Utils.colored_text("Error:", Color.RED), message)
+        colored_message = Utils.colored_text(f"Error: {message}", Color.RED)
+        logging.error(colored_message)
+        print(colored_message)
         _ = input(Utils.colored_text("\nPress Enter to continue...", Color.YELLOW))
 
     @staticmethod
     def success_msg(message: str) -> None:
-        print(Utils.colored_text("Success:", Color.GREEN), message)
+        colored_message = Utils.colored_text(f"Success: {message}", Color.GREEN)
+        logging.debug(message)  # Log the colored message
```

Comment:
Probably shouldn't have colors in the logging output, that will be _really_ hard to read in the file?
I'd argue that we should use a prefix and then the console logger handler could see that prefix and map it to the color instead.

Reply:
If you look, its sending the uncolored message to logging, and colored to print

oops.. i had fixed success but not error. Have now.

Reply:
Ahh, it was the comment that confused me.